Userland/Libraries/LibGfx/Bitmap.cpp

```diff
@@ -389,66 +389,111 @@ ErrorOr<NonnullRefPtr<Gfx::Bitmap>> Bitmap::scaled(float sx, float sy) const
+
+                int i = floorf(p);
+                int j = floorf(q);
+                float u = p - static_cast<float>(i);
+                float v = q - static_cast<float>(j);
+
+                auto a = get_pixel(i, j);
+                auto b = get_pixel(i + 1, j);
+                auto c = get_pixel(i, j + 1);
+                auto d = get_pixel(i + 1, j + 1);
+
+                auto e = a.interpolate(b, u);
+                auto f = c.interpolate(d, u);
+                auto color = e.interpolate(f, v);
+                new_bitmap->set_pixel(x, y, color);
+            }
+        }
+
+        // Bottom strip (excluding last pixel)
+        auto old_bottom_y = old_height - 1;
+        auto new_bottom_y = new_height - 1;
         for (int x = 0; x < new_width - 1; x++) {
             auto p = static_cast<float>(x) * static_cast<float>(old_width - 1) / static_cast<float>(new_width - 1);
-            auto q = static_cast<float>(y) * static_cast<float>(old_height - 1) / static_cast<float>(new_height - 1);
 
             int i = floorf(p);
-            int j = floorf(q);
             float u = p - static_cast<float>(i);
+
+            auto a = get_pixel(i, old_bottom_y);
+            auto b = get_pixel(i + 1, old_bottom_y);
+            auto color = a.interpolate(b, u);
+            new_bitmap->set_pixel(x, new_bottom_y, color);
+        }
+
+        // Right strip (excluding last pixel)
+        auto old_right_x = old_width - 1;
+        auto new_right_x = new_width - 1;
+        for (int y = 0; y < new_height - 1; y++) {
+            auto q = static_cast<float>(y) * static_cast<float>(old_height - 1) / static_cast<float>(new_height - 1);
+
+            int j = floorf(q);
             float v = q - static_cast<float>(j);
 
-            auto a = get_pixel(i, j);
-            auto b = get_pixel(i + 1, j);
-            auto c = get_pixel(i, j + 1);
-            auto d = get_pixel(i + 1, j + 1);
+            auto c = get_pixel(old_right_x, j);
+            auto d = get_pixel(old_right_x, j + 1);
 
-            auto e = a.interpolate(b, u);
-            auto f = c.interpolate(d, u);
-            auto color = e.interpolate(f, v);
-            new_bitmap->set_pixel(x, y, color);
+            auto color = c.interpolate(d, v);
+            new_bitmap->set_pixel(new_right_x, y, color);
         }
-    }
 
-    // Bottom strip (excluding last pixel)
-    auto old_bottom_y = old_height - 1;
-    auto new_bottom_y = new_height - 1;
-    for (int x = 0; x < new_width - 1; x++) {
-        auto p = static_cast<float>(x) * static_cast<float>(old_width - 1) / static_cast<float>(new_width - 1);
+        // Bottom-right pixel
+        new_bitmap->set_pixel(new_width - 1, new_height - 1, get_pixel(physical_width() - 1, physical_height() - 1));
 
-        int i = floorf(p);
-        float u = p - static_cast<float>(i);
+    } else if (old_height == 1 && old_width >= 1) {
```

Comment:
Nits: If you verify that the bitmap doesn't have any 0-length dimensions (which I assume shouldn't be possible), you're guaranteed here that (old_height > 1 || old_width > 1), so you can use early returns to guarantee you only need to check one condition here:

```
if (old_width > 1 && old_height > 1) {
    ...
    return;
}
if (old_height == 1) {
    // Horizontal strips
    return;
}
// Vertical strips
```

Reply:
Good suggestion, thank you.